src/arrayMethodSort.js

```diff
@@ -6,6 +6,52 @@
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
     // write code here
+    const isNumber = typeof this[0] === 'number';
+
+    if (compareFunction === undefined) {
+      this.forEach((element, index) => {
+        this[index] = element.toString();
+      });
```

Comment:
The compare function shouldn't change an array

Reply:
I think if I have no compare function as an argumet  I should compare my data like strings and then in the end cast my data to numbers.